provider/pkg/provider/util/testutil/test_ssh_server.go

```diff
@@ -0,0 +1,43 @@
+package testutil
+
+import (
+	"fmt"
+	"github.com/gliderlabs/ssh"
+	"github.com/stretchr/testify/require"
+	"net"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+type TestSshServer struct {
+	Host string
+	Port int64
+}
```

Comment:
I put this file in the testutil directory so it can be reused in the other test that depends on an SSH server.

I made that change here: https://github.com/theneva/pulumi-command/commit/bbae30e79444d8d28ad7a03c220c9cad9343d1da), and I'd love to include it as part of this PR. However, the test fails for me locally even on master, and I don't want to touch it (or block this PR on that unrelated change) since I can't verify the fix.

I'm happy to send that commit as a separate PR to pulumi/pulumi-command after this PR lands.

Reply:
That's a great move. I can help getting the other test to pass. It should pass on master in any case. What's the error you're getting?

Reply:
Just to be clear: The new test I wrote passes on my machine just fine. I hope we can land this PR without figuring out why the `TestOptionalLogging` test fails for me!

Buuuut since you asked:

I get assertions failures that seem to indicate that the output from the SSH server just doesn't get registered by the context.

I made a quick recording, showing me running `go test ./...` from `./provider` on the latest `origin/master` (914454482a2634938009eb41915467f867a6d4d6):

https://github.com/julsemaan/pulumi-command/assets/1404650/e471639c-bc0f-4416-9412-7716e44e37ab

I'm running this Go (installed with homebrew):

```
$ go version
go version go1.22.3 darwin/arm64
```

… on this machine:

<img width="392" alt="image" src="https://github.com/julsemaan/pulumi-command/assets/1404650/b3ec7981-ea04-41ff-9526-c109e94f3ede">

I also tried running `go install ./...` and `go mod tidy` just to be sure, but neither command did anything.

Again, I'm happy to help with figuring out why that test fails for me, but I'd really like to land this PR first 😅

Reply:
I can reproduce this locally. But if you run the same `go test ./...` in `provider/tests`, it works. That's how the CI build passes.

Reply:
I can reproduce this and will look into it.

Reply:
The regression was introduced by `36798ab Upgrade to latest pulumi-go-provider v0.17 (#442)`.

The reason CI tests pass is that it runs `go test` in `provider/tests`, not `/provider` - needs to be fixed as well.

Reply:
I see! That sounds like a fix for master. I'm happy to submit the change to use the new common test SSH server once it works on master

Reply:
This is now fixed in master.